contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -2232,3 +2233,80 @@ sp_addlinkedserver_internal(PG_FUNCTION_ARGS)
 
 	return (Datum) 0;
 }
+
+Datum
+sp_addlinkedsrvlogin_internal(PG_FUNCTION_ARGS)
+{
+	char *servername = PG_ARGISNULL(0) ? NULL : text_to_cstring(PG_GETARG_TEXT_P(0));
+	char *useself = PG_ARGISNULL(1) ? NULL : lowerstr(text_to_cstring(PG_GETARG_TEXT_P(1)));
+	char *username = PG_ARGISNULL(3) ? NULL : text_to_cstring(PG_GETARG_TEXT_P(3));
+	char *password = PG_ARGISNULL(4) ? NULL : text_to_cstring(PG_GETARG_TEXT_P(4));
```

Comment:
I missed this in initial revision, but why are we using PG_GETARG_TEXT_P instead of PG_GETARG_VARCHAR_PP?

Reply:
I think in this case it does not matter if we use PG_GETARG_VARCHAR_PP or PG_GETARG_TEXT_P; both do the exact same thing as far as I know.

Reply:
Not exactly! text_to_cstring is supposed to work with TEXT datatype. And, if you see implementation then also it is different from how we handle VARCHAR. So, we should use PG_GETARG_VARCHAR_PP for the consistency as VARCHAR datum might be of different form than TEXT.